src/xdsl/irdl.py

```diff
@@ -569,7 +569,9 @@ def irdl_op_definition(
             get_operand_or_result(self, idx, previous_vars, True))
         if isinstance(operand_def, VarOperandDef):
             previous_variadics += 1
-    if previous_variadics > 1 and AttrSizedOperandSegments() not in options:
+
+    if previous_variadics > 1 and AttrSizedOperandSegments(
+    ) not in options and not clsdict.get("parse"):
```

Comment:
Hmm, so I'm not entirely convinced that we want to entirely remove the segment size attributes.
The reason is that while having custom parsing is great, having a generic printer/parser is in my opinion essential, and this would require the `AttrSizedOperandSegment` to make sense.
What I think should happen, like in MLIR, is that the custom parsing function sets the attribute.